Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Controller.mo

```diff
@@ -0,0 +1,2260 @@
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate") if not have_priOnl and have_varPriPum
+     and have_floRegPri and not have_secFloSen
+    "Measured hot water flowrate through decoupler leg"
+    annotation (Placement(transformation(extent={{-440,-130},{-400,-90}}),
+      iconTransformation(extent={{-140,0},{-100,40}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSupSec(
+    final unit="K",
+    displayUnit="degC",
+    final quantity="ThermodynamicTemperature") if not have_priOnl and
+    have_varPriPum and have_temRegPri and have_priSecTemSen
+    "Measured hot water supply temperature in secondary circuit"
+    annotation (Placement(transformation(extent={{-440,-170},{-400,-130}}),
+      iconTransformation(extent={{-140,-40},{-100,0}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput TSupBoi[nBoi](
+    final unit=fill("K", nBoi),
+    displayUnit=fill("degC", nBoi),
+    final quantity=fill("ThermodynamicTemperature", nBoi)) if not have_priOnl
+     and have_varPriPum and have_temRegPri and not have_priSecTemSen
+    "Measured hot water supply temperatureat boiler outlets"
+    annotation (Placement(transformation(extent={{-440,-210},{-400,-170}}),
+      iconTransformation(extent={{-140,-70},{-100,-30}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput dpHotWatSec_rem[nSenSec](
+    final unit=fill("Pa", nSenSec),
+    displayUnit=fill("Pa", nSenSec),
+    final quantity=fill("PressureDifference", nSenSec)) if not
+    have_priOnl and have_varSecPum and (have_remDPRegSec or have_locDPRegSec)
+    "Measured differential pressure between hot water supply and return in secondary circuit"
+    annotation (Placement(transformation(extent={{-440,-250},{-400,-210}}),
+        iconTransformation(extent={{-140,-100},{-100,-60}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput dpHotWatPri_loc(
+    final unit="Pa",
+    displayUnit="Pa",
+    final quantity="PressureDifference") if have_varPriPum and have_locDPRegPri
+    "Measured differential pressure between hot water supply and return in primary circuit"
+    annotation (Placement(transformation(extent={{-440,-290},{-400,-250}}),
+        iconTransformation(extent={{-140,-130},{-100,-90}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput dpHotWatSec_loc(
+    final unit="Pa",
+    displayUnit="Pa",
+    final quantity="PressureDifference") if not have_priOnl and
+    have_varSecPum and have_locDPRegSec
+    "Measured differential pressure between hot water supply and return in secondary circuit"
+    annotation (Placement(transformation(extent={{-440,-330},{-400,-290}}),
+        iconTransformation(extent={{-140,-160},{-100,-120}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uHotWatIsoVal[nBoi](
+    final unit=fill("1",nBoi),
+    displayUnit=fill("1",nBoi)) if have_heaPriPum
+    "Measured boiler isolation valve position signals from plant"
```

Comment:
- This point should be optional per G36 4.11.1.
- If present, there should rather be 2 ***Boolean*** input points (DI) for open and closed end switch status, see G36 4.11.4.
- "Valve status" is not clear: valve open or closed end ***switch*** status (or position feedback if modulating) is clearer.
- That signal is used for fault detection and alarm only in G36. Here it is used by `Buildings.Controls.OBC.ASHRAE.PrimarySystem.BoilerPlant.Pumps.PrimaryPumps.Controller`. This seems incorrect based on:

  > 5.21.6.2. Enable lead primary hot water pump when any boiler isolation valve is ***commanded*** open. Disable the lead hot water pump when all boiler isolation valves are ***commanded*** closed.

  The signal that should be used is the command (yielded by subcontroller), not the feedback signal.